modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/EditSegmentsEntryDisplayContext.java

```diff
@@ -348,6 +350,41 @@ private JSONObject _getInitialQueryJSONObject(
 		FilterParser filterParser = _filterParserProvider.provide(
 			segmentsCriteriaContributor.getEntityModel());
 
+		if (GetterUtil.getBoolean(PropsUtil.get("feature.flag.LPS-166954"))) {
```

Comment:
Why this JSON should be returned only when the FF is set to true?

Reply:
Because if there is no Error Page for segments (and we have a FF for that) there shouldn't be a error to return

Reply:
The code inside the FF scope should include all the try catch part, because without it the `Expression expression = filterParser` sends an exception which is handled at another level and gets an global error